app/models/content_store_payload_version.rb

```diff
@@ -0,0 +1,41 @@
+class ContentStorePayloadVersion < ActiveRecord::Base
+  def self.increment(content_item_id:)
+    sql = <<-SQL
+        INSERT INTO content_store_payload_versions (content_item_id)
+        SELECT #{ content_item_id || "NULL" }
+        WHERE NOT EXISTS(
+        SELECT *
+        FROM content_store_payload_versions
+        WHERE content_item_id #{ content_item_id_sql(content_item_id) });
+
+        UPDATE content_store_payload_versions
+        SET current = COALESCE(current, 0) + 1
+        WHERE content_item_id #{ content_item_id_sql(content_item_id) }
+        RETURNING current;
+      SQL
+
+    ContentStorePayloadVersion.connection
+      .execute(sql)
```

Comment:
This isn't implicitly transacted. You probably want to ensure this is running in a transaction.